File: src/test/java/sx/LazyInstanceJavaTest.java
package sx;

import kotlin.jvm.functions.Function0;
import org.junit.Assert;
import org.junit.Test;

/**
 * Created by masc on 20/08/16.
 */
public class LazyInstanceJavaTest {
    class TestClass {
    }

    @Test
    public void testInitNull() {
        LazyInstance<TestClass> li = new LazyInstance<TestClass>(new Function0<TestClass>() {
            @Override
            public TestClass invoke() {
                return null;
            }
        });

        Assert.assertNull(li.get());
    }
}

File: src/test/java/sx/LazyInstanceJavaTest.kt
package sx

import org.junit.Assert
import org.junit.Test

/**
 * Created by masc on 20/08/16.
 */
class LazyInstanceJavaTest {
    internal inner class TestClass

    @Test
    fun testInitNull() {
        val li = LazyInstance({ null })

        Assert.assertNull(li.get())
    }
}
